src/pages/Menu/Components/ProductSpec/ProductSpec.js

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import Quantity from './Components/Quantity';
+import MenuDetail from './Components/MenuDetail';
+import Allergy from './Components/Allergy';
+import Country from './Components/Country';
+
+import API from '../../../../config';
+import './ProductSpec.scss';
+
+const Product = () => {
+  const [detail, setDetail] = useState();
+  const params = useParams();
```

Comment:
```suggestion
  const { id } = useParams();
```
Router에서 선언한 변수값만 객체로 받아오기 때문에 바로 구조분해 할당해서 사용하는게 더 가독성이 좋습니다.
아래에서 사용한 params.id 도 맞춰서 바꿔주세요.

Reply:
아핫 ..! 생각치도 못했습니당 !!!!